src/app/services/actions/autocomplete-action-creators.ts

```diff
@@ -0,0 +1,91 @@
+import { IAction } from '../../../types/action';
+import { IRequestOptions } from '../../../types/request';
+import { AUTOCOMPLETE_FETCH_ERROR, AUTOCOMPLETE_FETCH_PENDING, AUTOCOMPLETE_FETCH_SUCCESS } from '../redux-constants';
+
+export function fetchAutocompleteSuccess(response: object): IAction {
+  return {
+    type: AUTOCOMPLETE_FETCH_SUCCESS,
+    response,
+  };
+}
+
+export function fetchAutocompleteError(response: object): IAction {
+  return {
+    type: AUTOCOMPLETE_FETCH_ERROR,
+    response,
+  };
+}
+
+export function fetchAutocompletePending(): any {
+  return {
+    type: AUTOCOMPLETE_FETCH_PENDING
+  };
+}
+
+export function fetchAutoCompleteOptions(url: string): Function {
+  return async (dispatch: Function, getState: Function) => {
+    const devxApi = getState().devxApi;
+    const sampleQuery = getState().sampleQuery;
+    const headers = {
+      'Content-Type': 'application/json',
+    };
+
+    let permissionsUrl = `${devxApi}/openapi`;
+    permissionsUrl = `${permissionsUrl}?url=/${url}&style=geautocomplete&format=json`;
+
+    const options: IRequestOptions = { headers };
+
+    dispatch(fetchAutocompletePending());
+
+    try {
+      const response = await fetch(permissionsUrl, options);
+      if (response.ok) {
+        const autoCompleteOptions = await response.json();
+        const params = {
+          options: autoCompleteOptions,
+          url,
+          verb: sampleQuery.selectedVerb.toLowerCase()
+        };
+        const reduced = getReducedVersion(params);
+
+        return dispatch(fetchAutocompleteSuccess(reduced));
+      }
+
+      throw (response);
+
+    } catch (error) {
+      return dispatch(fetchAutocompleteError(error));
+    }
+  };
+}
+function getReducedVersion(params: any) {
+  const { options, url, verb } = params;
+  const { paths } = options;
+  try {
+    const parameters: any[] = [];
+    let rootPath = url;
+    if (url.includes('me/drive')) {
+      rootPath = url.substring(3);
```

Comment:
Why are we doings this only for the `me/drive` path?

Reply:
the me/drive path is the only one that doesn't follow the pattern in the url. It is a bug on the open api library that parses the data